src/errors.h

```diff
@@ -21,12 +21,12 @@ using namespace std::literals;
 
 namespace tasknow::errors {
 
-struct ApplicationError{
+struct ProtocolError{
     std::string message{};
 };
 
-struct UnrecoverableApplicationError: ApplicationError {};
-struct WarningApplicationError: ApplicationError {};
+struct UnrecoverableProtocolError: ProtocolError {};
+struct WarningProtocolError: ProtocolError {};
```

Comment:
A warning error is sure confusing. I'm not sure exceptions are a good mechanism to issue and process warnings.

Reply:
> A warning error is sure confusing.
Yep. I'm agree with you, but that's out of the scope of this PR.

> I'm not sure exceptions are a good mechanism to issue and process warnings.
You're right. But errors.h is not an exceptions. At least for now. I'm still thinking how to organize throwing exceptions and logging messages (info, errors, warnings, etc) in the best way.